editor/editor_properties.cpp

```diff
@@ -3651,6 +3652,21 @@ EditorProperty *EditorInspectorDefaultPlugin::get_editor_for_property(Object *p_
 					editor->set_save_mode();
 				}
 				return editor;
+			} else if (p_hint == PROPERTY_HINT_INPUT_NAME) {
+				EditorPropertyTextEnum *editor = memnew(EditorPropertyTextEnum);
+				Vector<String> options;
+				InputMap::get_singleton()->load_from_project_settings();
```

Comment:
You mean calling `InputMap::get_singleton()->load_from_project_settings();` ? it's the only way to get custom input actions here, unless there is another function that is better for this case.

Reply:
There's no reason to load it, and this will overwrite editor input map with project (and will ruin editor setting if you have default actions override in the project), and show editor actions in the list, so this is definitely the wrong way to do it. Instead, you can load the list directly, see `load_from_project_settings` implementation, something like:

```c++
	EditorPropertyTextEnum *editor = memnew(EditorPropertyTextEnum);
	Vector<String> options;
	List<PropertyInfo> pinfo;
	ProjectSettings::get_singleton()->get_property_list(&pinfo);

	for (const PropertyInfo &pi : pinfo) {
		if (!pi.name.begins_with("input/")) {
			continue;
		}

		String action_name = pi.name.substr(pi.name.find("/") + 1, pi.name.length());
		if ((p_hint_text != "allow_builtin") && InputMap::get_singleton()->get_builtins().has(action_name)) {
			continue;
		}
		options.append(action_name);
	}
	editor->setup(options, false, true);
	return editor;
```